src/player/Player.php

```diff
@@ -1507,7 +1507,7 @@ public function consumeHeldItem() : bool{
 			return true;
 		}
 
-		return false;
+		return true;
```

Comment:
This should not be changed because if this line was ever reached then it means the item was not consumed, and therefore it should return false

Reply:
The function returns false in case of unsuccessful food consumption, then to send the previous state of the item to the client (see InGamePaketHandler). Why send a network packet if the item is not food at all and there were no actions on the client side?

Reply:
because this handler isn't supposed to fire at all when food items are not used, and it's expected that it will only do so in the case where the client does something we don't expect it to. This expectation is broken because the use-item flag is set in wrong conditions.

Reply:
use-item flag should be set to false if the item is not used?

Reply:
that's what I've been trying to tell you the whole time.
> //TODO: check if item has a release action - if it doesn't, this shouldn't be set

Reply:
On stable, this bug manifests in a less obvious way: the famous arms-stuck-in-air bug, but the items are still usable because MobEquipmentPacket is handled unconditionally.

On the master branch, MobEquipmentPacket is ignored when the hotbar slot didn't change, which is why this bug appears on master and not on stable.

Reply:
ok, i have an idea how to implement this. Can I add a commit to this PR? Or do I need to open a new one?